allennlp/data/data_loaders/multitask_data_loader.py

```diff
@@ -0,0 +1,264 @@
+        _instances_, not _batches_, because if you're using batch size multipliers we don't know how
+        many batches the instances specified by the `scheduler` will turn out to be.
+    drop_last: `bool`, optional (default = `False`)
+        If this is `True`, we will not return the last batch if it is smaller than `batch_size`.
+        Note that this is kind of nonsensical to use if you're using `batch_size_multipliers`, as
+        you are not guaranteed to get an optimal packing, so you will likely have batches that don't
+        fill up the `batch_size` in that case, anyway.
+    num_workers: `Dict[str, int]`, optional (default = `None`)
+        Used when creating one `MultiProcessDataLoader` per dataset.  If you want non-default
+        behavior for this parameter in the `DataLoader` for a particular dataset, pass the
+        corresponding value here, keyed by the dataset name.
+    max_instances_in_memory: `Dict[str, int]`, optional (default = `None`)
+        Used when creating one `MultiProcessDataLoader` per dataset.  If you want non-default
+        behavior for this parameter in the `DataLoader` for a particular dataset, pass the
+        corresponding value here, keyed by the dataset name.
+    start_method: `Dict[str, str]`, optional (default = `None`)
+        Used when creating one `MultiProcessDataLoader` per dataset.  If you want non-default
+        behavior for this parameter in the `DataLoader` for a particular dataset, pass the
+        corresponding value here, keyed by the dataset name.
+    instance_queue_size: `Dict[str, int]`, optional (default = `None`)
+        Used when creating one `MultiProcessDataLoader` per dataset.  If you want non-default
+        behavior for this parameter in the `DataLoader` for a particular dataset, pass the
+        corresponding value here, keyed by the dataset name.
+    instance_chunk_size: `Dict[str, int]`, optional (default = `None`)
+        Used when creating one `MultiProcessDataLoader` per dataset.  If you want non-default
+        behavior for this parameter in the `DataLoader` for a particular dataset, pass the
+        corresponding value here, keyed by the dataset name.
+    """
+
+    def __init__(
+        self,
+        reader: MultiTaskDatasetReader,
+        data_path: Dict[str, str],
+        batch_size: int,
+        *,
+        scheduler: MultiTaskScheduler = None,
+        sampler: MultiTaskEpochSampler = None,
+        instances_per_epoch: int = None,
+        batch_size_multiplier: Dict[str, int] = None,
+        drop_last: bool = False,
+        num_workers: Dict[str, int] = None,
+        max_instances_in_memory: Dict[str, int] = None,
+        start_method: Dict[str, str] = None,
+        instance_queue_size: Dict[str, int] = None,
+        instance_chunk_size: Dict[str, int] = None,
+    ) -> None:
+        self.readers = reader.readers
+        self.data_paths = data_path
+        self.scheduler = scheduler
+        self.sampler = sampler
```

Comment:
`"homogeneous_roundrobin"` should be the default here, no?

Reply:
No, the default here should be `None`, because by default we use all instances.  This only should be set if `instances_per_epoch` is not `None`, and if you do that, it's better to make your configuration of how to sample explicit.